### ISSUE TITLE
Unchain into multiple statements for TYPES

### ISSUE THREAD

**Issue by jelliottp** [SAP/abap-cleaner#94]
I'd like to see this rule extended to work also for TYPES. Ideally we should unchain these just like constants, variables, methods, etc.

Before:
```
TYPES: BEGIN OF type1,
         field1 TYPE data,
       END OF type1,
       type2 TYPE STANDARD TABLE OF type1.
       BEGIN OF type3,
         field3 TYPE data,
       END OF type3.
       type3 TYPE STANDARD TABLE OF type3,
```

After:
```
TYPES: 
  BEGIN OF type1,
    field1 TYPE data,
  END OF type1.
TYPES type2 TYPE STANDARD TABLE OF type1.
TYPES: 
  BEGIN OF type3,
    field3 TYPE data,
  END OF type3.
TYPES type3 TYPE STANDARD TABLE OF type3.
```

**Comment by jmgrassau** [comment]
Hi Josh, 

yes, that absolutely makes sense, and guess which [comment](https://github.com/SAP/abap-cleaner/blob/823cbbc41ee63505922d87e41fc6b8ec2bac67bd/com.sap.adt.abapcleaner/src/com/sap/adt/abapcleaner/rules/declarations/ChainRule.java#L175C3-L175C3) I found inside the ChainRule code:

```Java
   // TODO: chains that just *contain* a BEGIN OF ... END OF section may nevertheless be partly unchained ...
   if (command.getFirstToken().matchesOnSiblings(true, TokenSearch.ASTERISK, "BEGIN OF"))
      return false;
```

So, currently, the rule skips any ```TYPES``` command that contains ```BEGIN OF``` anywhere, but it would be nice to unchain those, while keeping ```BEGIN OF ... END OF``` blocks together where possible (unless they are anyway interrupted by an annoying ```INCLUDE```…)

Kind regards,
Jörg-Michael

**Comment by AlexandreHT** [comment]
Hi, 
I must admit I prefer chaining structure and table declaration. The finest result would be to have an option for the following result

```ABAP
TYPES:  BEGIN OF ty_type1,
          field1 TYPE data,
        END OF ty_type1,
        tty_type1 TYPE STANDARD TABLE OF ty_type1.

TYPES:  BEGIN OF ty_type2,
          field3 TYPE data,
        END OF ty_type2,
        tty_type2 TYPE STANDARD TABLE OF ty_type2.
```
but I agree it complicates a bit the rule.

**Comment by jmgrassau** [comment]
Hi Josh, 

thanks for bringing this up! With the next release, "Unchain into multiple statements" will now also work on TYPES chains that contain BEGIN OF … END OF definitions, if they are not interrupted by an INCLUDE (sorry, that case would really complicate things too much, but I think most cases are simple cases):

![image](https://github.com/SAP/abap-cleaner/assets/79639298/761ed275-90fa-487a-9408-f6f6bf95036a)

@AlexandreHT: Nice point! For this, I added the following option, which is activated by default:
![image](https://github.com/SAP/abap-cleaner/assets/79639298/01c49e9a-cf62-433c-b782-087fb1dda390)

With that, you get:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/51493c0a-604a-4761-b9eb-7ba6eb66943c)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Josh, 

thanks for pointing this out – the enhancement (as described above) is now available in version 1.11.0 which was just released!

Kind regards,
Jörg-Michael
